lib/cloud_foundry.rb

```diff
@@ -11,6 +11,15 @@ def self.vcap_data
     end
   end
 
+  def self.app_url
+    if self.is_environment?
+      # (arbitrarily) use the shortest assigned route, if there are multiple
+      self.vcap_data['application_uris'].min_by(&:length)
+    else
+      nil
```

Comment:
I will open a separate style PR about this. My feeling on it as of right now is that it's a style I have not come across in the dozens of ruby apps I've worked on, so I am not crazy about it being a style we use in this app. But not going to hold this PR up over it as it can be discussed separately.